test/extended/dr/quorum_restore.go

```diff
@@ -50,6 +50,12 @@ var _ = g.Describe("[sig-etcd][Feature:DisasterRecovery][Disruptive]", func() {
 
 	oc := exutil.NewCLIWithoutNamespace("disaster-recovery")
 
+	// TODO(thomas): to be removed with a real test, this is just for testing the utility until then
+	g.It("[Feature:EtcdRecovery] should install ssh keys on CP nodes", func() {
+		err := installSSHKeyOnControlPlaneNodes(oc)
+		o.Expect(err).ToNot(o.HaveOccurred())
+	})
```

Comment:
not sure this might adversely impact the other serial jobs, but let me add this to the respective suites too. TRT will come after us anyway.

Reply:
interesting failure for the scaling test:
https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/27799/pull-ci-openshift-origin-master-e2e-aws-ovn-etcd-scaling/1638472477531181056

>             Message: "pods \"dr-ssh\" is forbidden: violates PodSecurity \"restricted:latest\": host namespaces (hostNetwork=true), privileged (container \"ssh-key-tester\" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container \"ssh-key-tester\" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container \"ssh-key-tester\" must set securityContext.capabilities.drop=[\"ALL\"]), runAsNonRoot != true (pod or container \"ssh-key-tester\" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container \"ssh-key-tester\" must set securityContext.seccompProfile.type to \"RuntimeDefault\" or \"Localhost\")",

do you know why there's a different security policy on the scaling jobs?

Reply:
okay I guess that was just a matter of using the oc cli in privileged mode

Reply:
Not entirely sure with what's going on with PodSecurity Admission (https://cloud.redhat.com/blog/pod-security-admission-in-openshift-4.11) but trying to create a pod locally in the ns `openshift-etcd-operator` with
```yaml
securityContext:
      privileged: true
```
succeeds although with a warning
```console
Warning: would violate PodSecurity "restricted:latest": privileged (container "nginx" must not set securityContext.privileged=true), allowPrivilegeEscalation != false (container "nginx" must set securityContext.allowPrivilegeEscalation=false), unrestricted capabilities (container "nginx" must set securityContext.capabilities.drop=["ALL"]), runAsNonRoot != true (pod or container "nginx" must set securityContext.runAsNonRoot=true), seccompProfile (pod or container "nginx" must set securityContext.seccompProfile.type to "RuntimeDefault" or "Localhost")
```

Trying the `openshift-etcd` namespace however works just fine with no warning since I know we already run the etcd containers as privileged on there.
The etcd namespace has the additional labels:
```
pod-security.kubernetes.io/warn=privileged
pod-security.kubernetes.io/enforce=privileged
```
so you could set those for openshift-etcd-operator so that namespace is unrestricted
https://kubernetes.io/docs/concepts/security/pod-security-standards/#privileged

And that's enough to work on 4.12 (where I'm testing) but on 4.13 or latest you might also need to set the label `security.openshift.io/scc.podSecurityLabelSync=false` as well. Otherwise that label would probably get set back.

Might want to read up on this to figure out what that's actually doing though :)

Reply:
> not sure this might adversely impact the other serial jobs, but let me add this to the respective suites too. TRT will come after us anyway.

Oh I just meant adding this snippet temporarily so we can piggyback those jobs to run our ssh-tester. We won't merge it into any of those suites.
We'll likely get jobs/presubmits created for the disruptive suite so we can run the quorum test on all/most platforms setup as a next step. But this is just a quick way to test this out across platforms right now.

Reply:
true, let me try to use the openshift-etcd ns first. I've tested this with a cluster bot before, so not sure why the admission policies are so wildly different :/

> security.openshift.io/scc.podSecurityLabelSync=false

okay I assume we have to patch those through the operator in https://github.com/openshift/cluster-etcd-operator/blob/master/bindata/etcd/ns.yaml?

Reply:
seems we got a little further now, but the test pod is timing out. I'm busy with the rhel stuff right now, so will get back after that...

Reply:
for the one job, the IP seems to not match anymore:
> testing SSH to [10.0.129.188]

from 
> NODE_IPS=( 10.0.129.188 10.0.167.235 10.0.253.176 )

yet etcd has completely different IPs set for its members:

```
{
    "header": {
        "cluster_id": 3579328861359447739,
        "member_id": 5018856643404254471,
        "raft_term": 9
    },
    "members": [
        {
            "ID": 3480470365665669229,
            "name": "ip-10-0-167-235.ec2.internal",
            "peerURLs": [
                "https://10.0.167.235:2380"
            ],
            "clientURLs": [
                "https://10.0.167.235:2379"
            ]
        },
        {
            "ID": 5018856643404254471,
            "name": "ip-10-0-143-100.ec2.internal",
            "peerURLs": [
                "https://10.0.143.100:2380"
            ],
            "clientURLs": [
                "https://10.0.143.100:2379"
            ]
        },
        {
            "ID": 5779358529620220134,
            "name": "ip-10-0-253-176.ec2.internal",
            "peerURLs": [
                "https://10.0.253.176:2380"
            ],
            "clientURLs": [
                "https://10.0.253.176:2379"
            ]
        }
    ]
}
```

and surprise, there's also no node matching those IPs. Very funny :/

Reply:
okay on Monday with a coffee it seems it was just one node not being there anymore, seems we got into conflict with our vertical scaling logic again ;)

Not sure where the node status was at that point, but I'll adjust the logic a little to find the right IPs.

Reply:
okay forget it, the CP nodes just come and go as they wish. By the time the tester pod runs, the pod that installs the key might be gone - so is the node. It's really not worth the hassle to hack those into the vertical scaling suite IMHO.

@hasbro17, let's get this merged? let's close this as part of this sprint, we can create a new test suite for the DR jobs and use that instead.

Reply:
Sounds good. Let's get a dedicated test setup for this with presubmits on more platforms.